README.md

```diff
@@ -16,3 +16,142 @@ bower install purescript-argonaut-core
 ## Documentation
 
 Module documentation is [published on Pursuit](http://pursuit.purescript.org/packages/purescript-argonaut-core).
+
+## Tutorial
+
+Some of Argonaut's functions might seem a bit arcane at first, so it can help
+to understand the underlying design decisions which make it the way it is.
+
+One approach for modelling JSON values would be to define an ADT, like this:
+
+```purescript
+data Json
+  = JNull
+  | JBoolean Boolean
+  | JArray (Array Json)
+  | JObject (StrMap Json)
+  [...]
+```
+
+And indeed, some might even say this is the obvious approach.
+
```

Comment:
The constructor names do correspond exactly to the real types which have those names, though, so I think it makes sense to use the same names. It does also say that Argonaut does not use this approach in the next paragraph.